platform/frontend_connectors/schema_transformer.go

```diff
@@ -1184,7 +1185,16 @@ func (s *SchemaCheckPass) Transform(plan *model.ExecutionPlan) (*model.Execution
 			{TransformationName: "BooleanLiteralTransformation", Transformation: s.applyBooleanLiteralLowering},
 		}...,
 	)
+	return transformationChain
+}
 
+func (s *SchemaCheckPass) Transform(plan *model.ExecutionPlan) (*model.ExecutionPlan, error) {
+
+	backendConnectorType := quesma_api.ClickHouseSQLBackend
+	if plan != nil && plan.BackendConnector != nil {
+		backendConnectorType = plan.BackendConnector.GetId()
+	}
```

Comment:
I think this whole logic above should be extracted to a separate func (~`determineBackendConnectorType(plan)`) and commented on why we have to do it this way.